src/coreclr/jit/instr.cpp

```diff
@@ -795,11 +811,69 @@ CodeGen::OperandDesc CodeGen::genOperandDesc(GenTree* op)
             assert(op->OperIsHWIntrinsic());
 
 #if defined(FEATURE_HW_INTRINSICS)
-            GenTreeHWIntrinsic* hwintrinsic = op->AsHWIntrinsic();
-            NamedIntrinsic      intrinsicId = hwintrinsic->GetHWIntrinsicId();
-
+            GenTreeHWIntrinsic* hwintrinsic  = op->AsHWIntrinsic();
+            NamedIntrinsic      intrinsicId  = hwintrinsic->GetHWIntrinsicId();
+            var_types           simdBaseType = hwintrinsic->GetSimdBaseType();
             switch (intrinsicId)
             {
+                case NI_AVX_BroadcastScalarToVector128:
+                case NI_AVX_BroadcastScalarToVector256:
+                {
+                    // we have the assumption that AVX_BroadcastScalarToVector*
+                    // only take the memory address as the operand.
+                    assert(hwintrinsic->isContained());
+                    assert(hwintrinsic->OperIsMemoryLoad());
+                    assert(hwintrinsic->GetOperandCount() == 1);
+                    assert(varTypeIsFloating(simdBaseType));
+                    GenTree* hwintrinsicChild = hwintrinsic->Op(1);
+                    assert(hwintrinsicChild->isContained());
+                    if (hwintrinsicChild->OperIs(GT_LCL_ADDR, GT_CLS_VAR_ADDR, GT_CNS_INT, GT_LEA))
+                    {
+                        addr = hwintrinsic->Op(1);
+                        break;
+                    }
+                    else
+                    {
+                        assert(hwintrinsicChild->OperIs(GT_LCL_VAR));
+                        return OperandDesc(simdBaseType, hwintrinsicChild);
+                    }
+                }
+
+                case NI_SSE3_MoveAndDuplicate:
+                case NI_AVX2_BroadcastScalarToVector128:
+                case NI_AVX2_BroadcastScalarToVector256:
+                case NI_AVX512F_BroadcastScalarToVector512:
+                {
+                    assert(hwintrinsic->isContained());
+                    if (intrinsicId == NI_SSE3_MoveAndDuplicate)
+                    {
+                        assert(simdBaseType == TYP_DOUBLE);
+                    }
+                    // If broadcast node is contained, should mean that we have some forms like
+                    // Broadcast -> CreateScalarUnsafe -> Scalar.
+                    // If so, directly emit scalar.
+                    // In the codes below, we specially handle the `Broadcast -> CNS_INT` form and
+                    // handle other cases recursively.
+                    GenTree* hwintrinsicChild = hwintrinsic->Op(1);
+                    assert(hwintrinsicChild->isContained());
```

Comment:
Hi @tannergooding, can you check if this assert is at the right place, I suppose if the broadcast operand should be contained by broadcast since this is the condition we previously made in lowering.